### DISCUSSION TITLE
ScriptReference/Application-wantsToQuit

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#137] [Notes]
# ScriptReference/Application-wantsToQuit

https://docs.unity3d.com/ScriptReference/Application-wantsToQuit.html

**Comment by vertxxyz** [comment]
This event is also run when clicking the application's close button [x] to quit, or when using `Alt+F4`.  

---
[`Environment.ExitCode`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.exitcode) is not set by [`Application.Quit`](https://docs.unity3d.com/ScriptReference/Application.Quit.html) before this function is called.  
If you want to read the exit code to determine whether to quit, manually assign it before calling `Quit`.
